packages/server/src/training/index.ts

```diff
@@ -40,92 +48,112 @@ export default class Trainer {
     });
   }
 
-  start(id: string, hyperparameters: unknown): void {
-    this.projects[id] = {
-      training_container: null,
-      metrics_container: null,
-      metrics: null
-    };
+  async start(id: string, hyperparameters: unknown): Promise<string> {
+    return new Promise((resolve, reject) => {
+      try {
+        const DATASET = path.basename(hyperparameters["datasetPath"]);
+        const MOUNT = Trainer.getMountPath(id).replace(/\\/g, "/");
+        const MOUNTCMD = Trainer.getMountCmd(MOUNT, this.CONTAINERMOUNT);
+        const CHECKPOINT = hyperparameters["checkpoint"];
 
-    const MOUNT = Trainer.getMountPath(id);
-    const CONTAINERMOUNT = "/opt/ml/model";
-    const DATASET = path.basename(hyperparameters["datasetPath"]);
+        this.projects[id] = {
+          name: hyperparameters["name"],
+          mount_path: MOUNT,
+          training_container: null,
+          metrics_container: null
+        };
 
-    let mountcmd = process.cwd();
-    if (mountcmd.includes(":\\")) {
-      // MOUNT PATH MODIFICATION IS FOR WINDOWS ONLY!
-      mountcmd = mountcmd.replace("C:\\", "/c/");
-      mountcmd = mountcmd.replace(/\\/g, "/");
-      console.log(mountcmd);
-    }
-    mountcmd = path.posix.join(mountcmd, MOUNT);
-    mountcmd = `${mountcmd}:${CONTAINERMOUNT}:rw`;
-
-    hyperparameters["batch-size"] = hyperparameters["batchSize"];
-    hyperparameters["eval-frequency"] = hyperparameters["evalFrequency"];
-    hyperparameters["dataset-path"] = hyperparameters["datasetPath"];
-    hyperparameters["percent-eval"] = hyperparameters["percentEval"];
+        hyperparameters["batch-size"] = hyperparameters["batchSize"];
+        hyperparameters["eval-frequency"] = hyperparameters["evalFrequency"];
+        hyperparameters["dataset-path"] = hyperparameters["datasetPath"];
+        hyperparameters["percent-eval"] = hyperparameters["percentEval"];
+        if (hyperparameters["checkpoint"] != "default") {
+          hyperparameters["checkpoint"] = path.posix.join("checkpoints", hyperparameters["checkpoint"]);
+        }
```

Comment:
The hyper parameters that the container expects should be their own type or interface that way we have a single source of truth if something changes.